test/e2e/build_test.go

```diff
@@ -116,6 +120,30 @@ func TestBuildSpecAndServe(t *testing.T) {
 	} else if cond.Status != corev1.ConditionTrue {
 		t.Fatalf("Build %q was not successful", buildName)
 	}
+
+	// Update the Configuration with an environment variable, which should trigger a new revision
+	// to be created, but without creating a new build.
+	if err := updateConfigWithEnvVars(clients, names, []corev1.EnvVar{{
+		Name:  "FOO",
+		Value: "bar",
+	}}); err != nil {
+		t.Fatalf("Failed to update config with environment variables: %v", err)
+	}
+
+	nextRevName, err := getNextRevisionName(clients, names)
+	if err != nil {
+		t.Fatalf("Error waiting for next revision to be created: %v", err)
+	}
+	names.Revision = nextRevName
+
+	nextRev, err := clients.ServingClient.Revisions.Get(nextRevName, metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to get latest Revision: %v", err)
+	}
+
+	if diff := cmp.Diff(rev.Spec.BuildRef, nextRev.Spec.BuildRef); diff != "" {
+		t.Fatalf("Unexpected differences in BuildRef: %v", diff)
+	}
 }
 
 func TestBuildAndServe(t *testing.T) {
```

Comment:
Should we add an explicit e2e test where updating the build spec triggers a new build?

Reply:
I'll add that now.